test/dotnet-core-uninstall.Tests/Shared/VSVersioning/VSVersionTests.cs

```diff
@@ -64,97 +77,167 @@ internal void TestGetUninstallable(string[] versions, bool[] allowed)
         [InlineData(new string[] { "5.0.0", "5.0.1", "10.100.100" }, new bool[] { false, false, false })]
         internal void TestGetUninstallableNonSdkVersionsWindows(string[] versions, bool[] allowed)
         {
-            TestGetUninstallableNonSdkVersions(versions, allowed);
+            var bundles = new List<Bundle>();
+            foreach (string v in versions)
+            {
+                bundles.Add(new Bundle<SdkVersion>(new SdkVersion(v), new BundleArch(), string.Empty, v));
+            }
+            TestGetUninstallableNonSdkVersions(bundles, allowed, null);
         }
 
         [MacOsOnlyTheory]
-        [InlineData(new string[] { "1.0.0", "1.0.1" }, new bool[] { true, true })]
-        [InlineData(new string[] { "1.0.0", "1.0.1", "1.1.0" }, new bool[] { true, true, true })]
-        [InlineData(new string[] { "5.0.0", "5.0.1", "10.100.100" }, new bool[] { true, true, true })]
-        internal void TestGetUninstallableNonSdkVersionsMac(string[] versions, bool[] allowed)
-        {
-            TestGetUninstallableNonSdkVersions(versions, allowed);
-        }
-
-        internal void TestGetUninstallableNonSdkVersions(string[] versions, bool[] allowed)
+        [InlineData(new string[] { "1.0.0" }, new bool[] { false }, new string[] { "1.0.0" }, new bool[] { false })]
+        [InlineData(new string[] { "1.0.0", "1.0.1" }, new bool[] { true, false }, new string[] { "1.0.0", "1.0.1" }, new bool[] { true, false })]
+        [InlineData(new string[] { "2.1.0", "1.0.1" }, new bool[] { false, true }, new string[] { "2.0.0", "1.1.0" }, new bool[] { false, false })]
```

Comment:
new string[] { "2.1.0", "1.0.1" }, new bool[] { false, true }

why 2.1.0 is allowed but 1.01 is not? Did we only want 3.0.0 for VS for mac?

Reply:
VS for mac only needs the highest SDK, so we only ever have to keep one. That test is saying that we should be able to uninstall 1.0.1 (it is allowed to be uninstalled) but 2.1.0 is required by VS for mac since it's the highest SDK "installed"